galloc/docs/optimisation.md

```diff
@@ -0,0 +1,320 @@
+
+Then the gas consumption was calculated as a percentage of total gas spent on the test case.
+
+We've intentionally did not put any info about testing machine, because we measure only gas consumption, which will be same for every machine, since we have fixed weights for every instruction and syscall.
+
+The test cases were:
+- `NFT init -> mint -> burn`: This test case is for measuring the performance of the NFT contract, which is the one of the common cases of smart contracts. The test case consists of the following steps:
+  - `init`: Initialise the NFT contract.
+  - `mint`: Mint 1 NFT.
+  - `burn`: Burn 1 NFT.
+
+  This is quite simple case and it's not the most common case, but it's the most simple case we can think of.
+
+  You can see the code of the test case [here](https://github.com/gear-dapps/non-fungible-token/blob/0.2.10/tests/node_tests.rs) (`burn-test`).
+- `FT stress-test` with <kbd>gtest</kbd>: This test case is for measuring the performance of the FT contract, which is also the one of the common cases of smart contracts. The test case consists of the following steps:
+  - `init`: Initialise the FT contract.
+  - `mint`: Mint 1 000 000 FT to the first user.
+  - `transfer`: Transfer 6 000 FT to the first 20 accounts from first user.
+  - `balance`: Check the balance of first account to prove it sent 6 000 FT to the first 20 accounts.
+  - `balance`: Check the balance of first 20 accounts.
+  - `transfer`: Transfer 1 000 FT from the first 20 accounts to users 21-100.
+  - `balance`: Check the balance of first 20 accounts to prove it sent 1 000 FT to users 21-100.
+  - `balance`: Check the balance of users 21-100 to prove they've received 1 000 FT from the first 20 accounts.
+  - `mint`: Mint 1 000 000 FT to the second user. 
+  - `mint`: Mint 5 000 FT to users 87..120.
+  - `balance`: Check the balance of users 87..120 to prove they've received 5 000 FT after mint.
+  - `transfer`: Transfer 1 000 FT from users 87..120 to the first user.
+  - `total_supply`: Check the total supply of FT to prove it's sums up to correct value.
+  - `balance`: Check the balance of users 918..13400
+  - `mint`: Mint 5 000 FT to users 918..13400.
+  - `transfer`: Transfer `i / 4` FT from users 918..13400 to the user `i * 2`.
+
+  The next two steps were repeated 30 times:
+  - `balance`: Check the balance of users 1..130
+  - `transfer`: Transfer 1 FT from users 1..130 to the first user.
+
+  This test tries to imitate what happens when the FT contract is used in the real world, with many clients and memory allocations therefore.
+
+- `FT stress-test` with <kbd>gclient</kbd>: This test case has the same steps as the previous one, but it's executed with <kbd>gclient</kbd> instead of <kbd>gtest</kbd>. This test is most similar to the real world case, because it's executed with the real node and will give the most real gas usage. 
```

Comment:
```suggestion
- `FT stress-test` with <kbd>gclient</kbd>: This test case has the same steps as the previous one, but it's executed with <kbd>gclient</kbd> instead of <kbd>gtest</kbd>. This test is most similar to the real world case because it's executed with the real node and will give the most realistic gas usage. 
```